x2gbfs/providers/noi.py

```diff
@@ -0,0 +1,124 @@
+        pass
+
+    def count_car(self, station_id, type):
+        if self.cars.get(station_id) is None:
+            self.cars[station_id] = { type: 1 }
+        elif self.cars[station_id].get(type) is None:
+            self.cars[station_id][type] = 1
+        else:
+            count = self.cars[station_id][type]
+            self.cars[station_id][type] = count + 1
+
+    def total_cars(self, station_id):
+        cars = self.cars.get(station_id)
+        if cars is None:
+            return 0
+        else:
+            count = 0
+            for i in cars:
+                count = count + cars[i]
+        return count
+
+class NoiProvider(BaseProvider):
+    STATION_URL = "https://mobility.api.opendatahub.com/v2/flat%2Cnode/CarsharingStation?limit=500&offset=0&shownull=false&distinct=true"
+    CAR_URL = "https://mobility.api.opendatahub.com/v2/flat%2Cnode/CarsharingCar?limit=500&offset=0&shownull=false&distinct=true"
+
+    def __init__(self):
+        pass
+
+    def load_vehicles(self, default_last_reported: int) -> Tuple[Optional[Dict], Optional[Dict]]:
+        response = requests.get(self.CAR_URL, timeout=20)
+        raw_cars = response.json()["data"]
+        types = {}
+        for i in raw_cars:
+
+                id = self.extract_type_id(i)
+                types[id] = {
+                    # See https://github.com/MobilityData/gbfs/blob/v2.3/gbfs.md#vehicle_typesjson
+                    'vehicle_type_id': id,
+                    'form_factor': 'car',
+                    'propulsion_type': self.extract_propulsion(i),
+                    'max_range_meters': 500000,
+                    'name': i["smetadata"]["brand"].strip(),
+                    'wheel_count': 4
+                }
+        return types, {}
+
+    def extract_type_id(self, i):
+        output = i["smetadata"]["brand"].lower().strip().replace("!", "")
+        output =  re.sub(r'[^a-z0-9]+', '-', output)
+        output =  re.sub(r'-$', '', output)
+        return output
+
+    def extract_propulsion(self, i):
+        if(self.extract_type_id(i).__contains__("elektro")):
+            return "electric"
+        else:
+            return "combustion"
+
+
+    def load_stations(self, default_last_reported: int) -> Tuple[Optional[Dict], Optional[Dict]]:
+
+        response = requests.get(self.STATION_URL, timeout=20)
+        raw_stations = response.json()["data"]
+
+        info = {}
+        for i in raw_stations:
+            id = i["scode"]
+            coord = i["scoordinate"]
+            info[id] = {
+                "station_id": id,
+                "name": i["sname"],
+                "lon": coord["x"],
+                "lat": coord["y"],
+            }
+
+        status = self.load_status(default_last_reported)
+        return info, status,
+
+    def load_status(self, last_reported: int) -> Optional[Dict]:
+        response = requests.get(self.CAR_URL, timeout=20)
+        raw_cars = response.json()["data"]
+        infos = {}
```

Comment:
better name this status, to avoid confusion with the static info

Reply:
93f498f34d5db56981d811aef90a3525bd0c694c